### PR TITLE
add member model and its test

### PR DESCRIPTION
**Pull request by zhang-shengping**

### CODE REVIEW

File: app/waf/src/controllers/member.controller.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
+// Copyright F5 Networks, Inc. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
```
Comment (zhaoqin-github) [inline app/waf/src/controllers/member.controller.ts]: Apache 2.0 license.

File: app/waf/src/models/member.model.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Copyright F5 Networks, Inc. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
```
Comment (zhaoqin-github) [inline app/waf/src/models/member.model.ts]: Apache 2.0 license.

File: app/waf/src/controllers/member.controller.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+const prefix = '/adcaas/v1';
+
+export class MemberController {
+  constructor(
+    @repository(MemberRepository)
+    public memberRepository: MemberRepository,
+  ) {}
+
+  @post(prefix + '/members', {
```
Comment (zhaoqin-github) [inline app/waf/src/controllers/member.controller.ts]: /adcaas/v1/pools/{pool_id}/members

Comment (zhaoqin-github) [reply app/waf/src/controllers/member.controller.ts]: Also need to update pool's member list to add this member to a pool.

File: app/waf/src/controllers/pool.controller.ts [hunk 4]
```diff
@@ -136,4 +140,141 @@ export class PoolController {
...
     await this.poolRepository.deleteById(id);
   }
+
+  /**
+   * These member control functions are moved to pool controller,
+   * it is because of this issues:
+   * https://github.com/strongloop/loopback/issues/4142
+   */
+
+  @get(prefix + '/members/count', {
```
Comment (zhaoqin-github) [inline app/waf/src/controllers/pool.controller.ts]: We may remove this handler, but I am ok to merge it now.

File: app/waf/src/controllers/pool.controller.ts [hunk 5]
```diff
@@ -136,4 +140,141 @@ export class PoolController {
...
+      },
+    },
+  })
+  async countMembers(
+    @param.query.object('where', getWhereSchemaFor(Member)) where?: Where,
+  ): Promise<Count> {
+    return await this.memberRepository.count(where);
+  }
+
+  @get(prefix + '/members', {
```
Comment (zhaoqin-github) [inline app/waf/src/controllers/pool.controller.ts]: We may remove this handler, but I am ok to merge it now.

**Comment by zhaoqin-github** [comment]
Two exception paths are not covered by UT. However, we may select a DAO style of implementation and rewrite the code soon. Let us fix that in near future.

File: app/waf/src/controllers/pool.controller.ts [hunk 6]
```diff
@@ -136,4 +140,141 @@ export class PoolController {
...
+    },
+  })
+  async getMembers(
+    @param.path.string('pool_id') pool_id: string,
+  ): Promise<Member[]> {
+    let pool: Pool = await this.poolRepository.findById(pool_id);
+    if (pool.members) {
+      let members: Member[] = [];
+      for (let mbr of pool.members) {
+        members.push(await this.memberRepository.findById(mbr));
```
Comment (zongzw) [inline app/waf/src/controllers/pool.controller.ts]: Why not use memberRepository.find(filter); to return the array directly??

**Pull request merged**